### ISSUE TITLE
ERROR in Symbol MatButtonToggleGroup declared in...

### ISSUE THREAD

**Issue by adamraetz** [angular/angular#35611]
# 🐞 bug report

### Affected Package

This is issue is caused by the updating out Angular project from 8 to 9. 

### Is this a regression?

There were no issues like this when running ng build in Angular 8.
 

### Description
After updating all relevant packages and trying to run  "ng build" or "ng serve"

 

## 🔬 Minimal Reproduction

Updating to Angular 9 from 8.

I cannot recreate in Stackblitz. And since it's enterprise system I cannot share a repo.

## 🔥 Exception or Error
<pre><code>

ERROR in Symbol MatButtonToggleGroup declared in
 C:/Users/asraetz/source/repos/XXXXXXXX/ClientApp/node_modules/@angular/material/button-toggle/button-toggle.d.ts 
is not exported from @angular/material/button-toggle/button-toggle-module 
(import into C:/Users/asraetz/source/repos/XXXXXXXXX/ClientApp/src/app/app.component.ts)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 9.0.1
Node: 12.10.0
OS: win32 x64

Angular: 9.0.1
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, material, platform-browser
... platform-browser-dynamic, platform-server, router
Ivy Workspace: Yes

Package                                    Version
--------------------------------------------------------------------
@angular-devkit/architect                  0.900.2
@angular-devkit/build-angular              0.900.2
@angular-devkit/build-optimizer            0.900.2
@angular-devkit/build-webpack              0.900.2
@angular-devkit/core                       9.0.2
@angular-devkit/schematics                 9.0.1
@angular/flex-layout                       9.0.0-beta.29
@angular/http                              7.2.15
@ngtools/webpack                           9.0.2
@nguniversal/module-map-ngfactory-loader   8.2.6
@schematics/angular                        9.0.1
@schematics/update                         0.900.1
rxjs                                       6.5.4
typescript                                 3.7.5
webpack                                    4.41.2
</code></pre>

**Anything else relevant?**

I am using a CustomeMaterialsModule.ts, and all modules are not giving any errors for the import paths.
<pre><code>
import { NgModule } from '@angular/core';
import { FlexLayoutModule } from '@angular/flex-layout';

import { MatAutocompleteModule } from '@angular/material/autocomplete';
import { MatButtonModule } from '@angular/material/button';
import { MatButtonToggleModule } from '@angular/material/button-toggle/button-toggle-module';
import { MatCardModule } from '@angular/material/card';
import { MatCheckboxModule } from '@angular/material/checkbox/checkbox-module';
import { MatChipsModule } from '@angular/material/chips';
import { MatNativeDateModule } from '@angular/material/core/datetime';
import { MatRippleModule } from '@angular/material/core/ripple';
import { ErrorStateMatcher, ShowOnDirtyErrorStateMatcher } from '@angular/material/core/error/error-options';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatDialogModule } from '@angular/material/dialog/dialog-module';
import { MatDividerModule } from '@angular/material/divider/divider-module';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatGridListModule } from '@angular/material/grid-list';
import { MatIconModule } from '@angular/material/icon/icon-module';
import { MatInputModule } from '@angular/material/input/input-module';
import { MatListModule } from '@angular/material/list';
import { MatMenuModule } from '@angular/material/menu';
import { MatPaginatorModule } from '@angular/material/paginator/paginator-module';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatRadioModule } from '@angular/material/radio';
import { MatSelectModule } from '@angular/material/select/select-module';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatSlideToggleModule } from '@angular/material/slide-toggle';
import { MatSliderModule } from '@angular/material/slider';
import { MatSnackBarModule } from '@angular/material/snack-bar/snack-bar-module';
import { MatSortModule } from '@angular/material/sort/sort-module';
import { MatStepperModule } from '@angular/material/stepper';
import { MatTableModule } from '@angular/material/table/table-module';
import { MatTabsModule } from '@angular/material/tabs';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatTooltipModule } from '@angular/material/tooltip/tooltip-module';
import { DigitOnlyModule } from '@uiowa/digit-only';
import { ScrollingModule } from '@angular/cdk/scrolling/scrolling-module';

const Materials = [

    FlexLayoutModule,

    // Material
    MatAutocompleteModule,
    MatButtonModule,
    MatButtonToggleModule,
    MatCardModule,
    MatCheckboxModule,
    MatChipsModule,
    MatDatepickerModule,
    MatDialogModule,
    MatDividerModule,
    MatExpansionModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatListModule,
    MatMenuModule,
    MatNativeDateModule,
    MatPaginatorModule,
    MatProgressBarModule,
    MatProgressSpinnerModule,
    MatRadioModule,
    MatRippleModule,
    MatSelectModule,
    MatSidenavModule,
    MatSlideToggleModule,
    MatSliderModule,
    MatSnackBarModule,
    MatSortModule,
    MatStepperModule,
    MatTableModule,
    MatTabsModule,
    MatToolbarModule,
    MatTooltipModule,
    ScrollingModule,
    DigitOnlyModule,
];

@NgModule({
  imports: Materials,
    exports: Materials,
  providers: [
    { provide: ErrorStateMatcher, useClass: ShowOnDirtyErrorStateMatcher }
  ]
})
export class CustomMaterialModule { }
</code></pre>

It's being imported into app_module.ts
<pre><code>
import { BrowserModule } from '@angular/platform-browser';
import { NgModule, ErrorHandler } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { AppComponent } from './app.component';
import { AppRoutingModule } from './app-routing.module';
import { PageNotFoundComponent } from './infrastructure/not-found.component';
import { CustomMaterialModule } from './infrastructure/custom-material-module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MenuModule } from './menu/menu.module';
import { ToastrModule } from 'ngx-toastr';
import { PendingChangesConfirmDialog } from './infrastructure/pending-changes.guard';
import { GlobalErrorHandler } from './infrastructure/global-error-handler.service';
import { NgxMaskModule, IConfig } from 'ngx-mask';
export const options: Partial<IConfig> | (() => Partial<IConfig>) = {};

import { MsalModule } from '@azure/msal-angular';
import { OAuthSettings } from '../oauth'
import { MatTableExporterModule } from 'mat-table-exporter';
import { BreadcrumbComponent } from './breadcrumb/breadcrumb.component';
import { BreadcrumbModule } from 'primeng/breadcrumb';

const isIE = window.navigator.userAgent.indexOf("MSIE ") > -1 || window.navigator.userAgent.indexOf("Trident/") > -1;

@NgModule({
    declarations: [
        AppComponent,
        PageNotFoundComponent,
        PendingChangesConfirmDialog,
        BreadcrumbComponent,
    ],
    imports: [
        BrowserModule.withServerTransition({ appId: 'ng-cli-universal' }),
        BrowserAnimationsModule,
        AppRoutingModule,
        MenuModule,
        MatTableExporterModule,
        HttpClientModule,
        FormsModule,
        ReactiveFormsModule,
        MsalModule.forRoot({
            clientID:  OAuthSettings.appId,
            authority: OAuthSettings.authority,
            validateAuthority: true,
            cacheLocation: 'localStorage',
            storeAuthStateInCookie: isIE,
            popUp: false,
            consentScopes: ['user.read'],
        }),
        ToastrModule.forRoot({
            timeOut: 10000, positionClass: 'toast-top-full-width', preventDuplicates: true
        }),
        NgxMaskModule.forRoot(options),
        CustomMaterialModule,
        BreadcrumbModule
    ],

    providers: [{ provide: ErrorHandler, useClass: GlobalErrorHandler }],
    bootstrap: [AppComponent],
    entryComponents: [],
})
export class AppModule { }
</code></pre>

app.component.ts
<pre><code>
import { Component, OnInit, Inject } from '@angular/core';
import { UserInfoService } from './infrastructure/userinfo.service';
import { BroadcastService, MsalService } from '@azure/msal-angular';
import { Subscription } from 'rxjs';
import { PageHeaderService, PageHeaderModel } from './infrastructure/pageheader.service.service';

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit {

    public isIframe: boolean;
    loggedIn: boolean;
    private pageHeaderSubscription: Subscription;
    private userName: string;

    public pageHeaderName: string = "Home";
    public pageHeaderMenuItemId: number;
    public pageHeaderShowPin: boolean;

    constructor(private msalService: MsalService,
        private broadcastService: BroadcastService,
        private uiService: UserInfoService,
        private phService: PageHeaderService,
        @Inject('BASE_URL') private baseUrl: string) {

        //set the client id
        this.SetClientId();

        this.isIframe = window !== window.parent && !window.opener;

        let user = this.msalService.getUser();
        
        if (user) {
            this.loggedIn = true;
            this.userName = user.displayableId.split("@")[0].toLowerCase();
            this.uiService.loadUserInfo(this.userName);
        } else {
            this.loggedIn = false;
        }

        //subscribe to the page header changes
        this.pageHeaderSubscription = this.phService.pageHeaderObservable.subscribe((data: PageHeaderModel) => {
            this.pageHeaderName = data.name;
            this.pageHeaderMenuItemId = data.menuItemId;
            this.pageHeaderShowPin = data.showPin;
        });

    }

    ngOnInit(): void {

        console.log("IN App Component!");

        //Single Sign On
        this.broadcastService.subscribe("msal:loginFailure", (payload) => {
            alert("login failure " + JSON.stringify(payload));
            this.loggedIn = false;
        });

        this.broadcastService.subscribe("msal:loginSuccess", () => {
            this.loggedIn = true;
            this.uiService.loadUserInfo(this.userName);
        });

        if (!this.loggedIn) {
            this.login();
        }

    }

    ngOnDestroy() {

        this.broadcastService.getMSALSubject().next(1);

        if (this.pageHeaderSubscription) {

            this.pageHeaderSubscription.unsubscribe();

        }

    }

    //log the user into the msal service.
    login() {

        this.msalService.loginRedirect();

    }

    //set the client id for the base url
    private SetClientId() {
        switch (this.baseUrl) {
            //Dev
            case `https://anyURL.com/`:
                this.msalService.clientId = 'XXXXXXXX';
                break;
            //Test
            case `https://anyURL.com/`:
                this.msalService.clientId = 'XXXXXX';
                break;
            //Prod
            case `https://anyURL.com/`:
                this.msalService.clientId = 'XXXXXXX';
                break;
            //Local
            default:
                this.msalService.clientId = 'XXXXXX';
        }
    }

}

</code></pre>

**Comment by alexzuza** [comment]
Can you please add the code of your `app.component.ts` file?

**Comment by adamraetz** [comment]
> Can you please add the code of your `app.component.ts` file?

Added into the main post. Sorry about that had it ready to paste in and didn't.

**Comment by alexzuza** [comment]
@adamraetz I see lots of wrong deep imports, e.g

```
import { MatButtonToggleModule } from '@angular/material/button-toggle/button-toggle-module';
          || should be
          \/
import { MatButtonToggleModule } from '@angular/material/button-toggle';




import { MatCheckboxModule } from '@angular/material/checkbox/checkbox-module';

                ||
               \/
import { MatCheckboxModule } from '@angular/material/checkbox';

```

and many others. You should import all identifiers from secondary entry points

`@angular/material/{secondaryEntryPoint}`

**Comment by adamraetz** [comment]
> @adamraetz I see lots of wrong deep imports, e.g
> 
> ```
> import { MatButtonToggleModule } from '@angular/material/button-toggle/button-toggle-module';
>           || should be
>           \/
> import { MatButtonToggleModule } from '@angular/material/button-toggle';
> 
> 
> 
> 
> import { MatCheckboxModule } from '@angular/material/checkbox/checkbox-module';
> 
>                 ||
>                \/
> import { MatCheckboxModule } from '@angular/material/checkbox';
> ```
> 
> and many others. You should import all identifiers from secondary entry points
> 
> `@angular/material/{secondaryEntryPoint}`
> 
> Online reproduction of your issue https://ng-run.com/edit/gGfnpByJez6JohfkYBwp

Thanks Alex. I thought Angular/Material 9 required explicit import pathing? 

I am not sure what has changed, maybe my material version was out of sync with the core version, but even this morning if I imported per your instructions, the Modules would have errors associated with them until I specified the exact path. 

I appreciate the help. 

Now I'll be putting together another issue because while it now builds I am getting a:

`Error: Cannot instantiate cyclic dependency! ApplicationRef
    at throwCyclicDependencyError (core.js:7986)
    at R3Injector.hydrate (core.js:16761)
    at R3Injector.get (core.js:16526)
    at injectInjectorOnly (core.js:905)
    at Module.ɵɵinject (core.js:915)
    at Object.Overlay_Factory [as factory] (ngx-toastr.js:451)
    at R3Injector.hydrate (core.js:16765)
    at R3Injector.get (core.js:16526)
    at injectInjectorOnly (core.js:905)
    at Module.ɵɵinject (core.js:915)`

In the browser and it is stuck on our app loading screen.

Thank you again.

**Comment by alexzuza** [comment]
> Error: Cannot instantiate cyclic dependency! ApplicationRef 

@adamraetz I think it's a duplicate of https://github.com/angular/angular/issues/35225

**Comment by AndrewKushnir** [comment]
Hi @adamraetz,

As @alexzuza mentioned, the error message looks very similar to the one from the #35225. However the problem in #35225 was related to unused dependency in `Router` that was removed in a PR (that will be merged soon). Based on the error that I see in your message, it's related to `Overlay` component, so the fix for #35225 may not resolve the problem that you refer to. Could you please create a minimum repro with the use case that you have, so that we can look into it?

Thank you.

**Comment by pkozlowski-opensource** [comment]
I'm going to close this one for now as we haven't heard back with the reproduce scenario. Please open a new issue if / when you can provide one.
